src/sql/parts/connection/common/connectionManagementService.ts

```diff
@@ -1366,4 +1366,17 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			});
 		});
 	}
+
+	/**
+	 * Serialize connection with options provider
+	 */
+	public serializeConnectionString(connectionString: string, provider?: string): Thenable<sqlops.ConnectionInfo> {
+		if (provider) {
+			return this._providers.get(provider).onReady.then(e => {
+				return e.serializeConnectionString(connectionString);
+			});
+		} else {
+			return Promise.resolve(undefined);
```

Comment:
Why is provider optional if it's going to fail if you don't give one?

Reply:
I intended in the future serializing a connection string could be a map reduce type of operation, and it would return which provider/s were able to serialize it. Obviously thats out of scope, so I'll remove it and add a note somewhere.